src/api/query.ts

```diff
@@ -35,22 +38,23 @@ export const timeRangeSQLCondition = (timePartitionColumn: string, startTime: Da
 };
 
 export const formQueryOpts = (logsQuery: FormQueryOptsType) => {
-	const { startTime, endTime, streamName, limit, pageOffset, timePartitionColumn = 'p_timestamp' } = logsQuery;
-	const optimizedStartTime = optimizeTime(startTime);
-	const optimizedEndTime = optimizeTime(endTime);
-	const orderBy = `ORDER BY ${timePartitionColumn} desc`;
-	const timestampClause = timeRangeSQLCondition(timePartitionColumn, optimizedStartTime, optimizedEndTime);
-	const offsetPart = _.isNumber(pageOffset) ? `OFFSET ${pageOffset}` : '';
-	const query = `SELECT * FROM \"${streamName}\" where ${timestampClause} ${orderBy} ${offsetPart} LIMIT ${limit} `;
-	return { query, startTime: optimizedStartTime, endTime: optimizedEndTime };
+	const queryBuilder = new QueryBuilder(logsQuery);
```

Comment:
Why we did a class implementation ? The created instance was never passed outside this function. So there is no need for a class here.

Give a read on Separation of concern.

Reply:
Problem we want to solve is that earlier there was this useTrino flag passed around everywhere which is error prone. How do you propose to fix that?

Reply:
Oh ok then.

Reply:
I am more than happy if you can propose a better / idiomatic approach on how to do this.

Reply:
1. Even now, we are sending the 'queryEngine' in place of 'useTrino'. So no difference.
2. App initializes only one time. We should assume the query engine the first time itself and save it as a boolean (useTrino: boolean, since we are not going to include any other connector anytime soon) in the AppProvider. So we dont have to compare the string everytime, everywhere. 
3. trinoquery & query endpoint definition should remain separate. 

Stating all the above, we dont really need a class. Instead of passing the string everywhere, pass just the boolean since thats what we need at the end. Revert back all the changes and just set and set the useTrino (boolean) in the app provider.

Reply:
And 'queryEngine' shouldnt be query_engine from the backend ?

Reply:
> Instead of passing the string everywhere, pass just the boolean since thats what we need at the end.

We have to avoid this. It has to dynamically decide which engine to use based on API response.

Reply:
Thats a very valid point if we are going to add more connectors.

Reply:
Right now there is one, but we have to make it modular so we can extend as needed

Reply:
about the endpoint def.  1 function to dynamically decide or separate endpoints for every connectors?

Reply:
If we are going to include more connectors:

Then the class should provide everything for making a request.
All the logic related to query engine should be inside the class. Define separate urls for separate engines and the class should decide and return the url also.